modules/core/src/test/java/org/locationtech/jts/algorithm/AngleTest.java

```diff
@@ -97,9 +101,52 @@ public void testNormalize() throws Exception
 		assertEquals(Angle.normalize(2.5*Math.PI), 0.5*Math.PI, TOLERANCE);
 		assertEquals(Angle.normalize(3*Math.PI), Math.PI, TOLERANCE);	
 		assertEquals(Angle.normalize(4 * Math.PI), 0.0, TOLERANCE);
-		
-		
   }
 
+  public void testInteriorAngle() {
+		Coordinate p1 = new CoordinateXY(1, 2);
+		Coordinate p2 = new CoordinateXY(3, 2);
+		Coordinate p3 = new CoordinateXY(2, 1);
+
+		// Tests all interior angles of a triangle "POLYGON ((1 2, 3 2, 2 1, 1 2))"
+		assertEquals(45, Math.toDegrees(Angle.interiorAngle(p1, p2, p3)), 0.01);
+		assertEquals(90, Math.toDegrees(Angle.interiorAngle(p2, p3, p1)), 0.01);
+		assertEquals(45, Math.toDegrees(Angle.interiorAngle(p3, p1, p2)), 0.01);
+		// Tests interior angles greater than 180 degrees
+		assertEquals(315, Math.toDegrees(Angle.interiorAngle(p3, p2, p1)), 0.01);
+		assertEquals(270, Math.toDegrees(Angle.interiorAngle(p1, p3, p2)), 0.01);
+		assertEquals(315, Math.toDegrees(Angle.interiorAngle(p2, p1, p3)), 0.01);
+  }
 
+  /**
+   * Tests interior angle calculation using a large number of random triangles
+   */
+  public void testInteriorAngle_randomTriangles() {
```

Comment:
I would rather not have lengthy tests like this in the main set of unit tests, to avoid impacting CI performance.

You could reduce the number of tests to say 100 - it can be increased manually if needed.

Reply:
Reduced to 100.